core/common.h

```diff
@@ -359,6 +359,8 @@ static inline void mbedtls_xor_no_simd(unsigned char *r,
  */
 #if defined(static_assert) && !defined(__FreeBSD__)
 #define MBEDTLS_STATIC_ASSERT(expr, msg)    static_assert(expr, msg)
+#elif defined(__GNUC__) && ((__GNU__ == 4 && __GNUC_MINOR >= 6) || __GNU__ > 4) && !defined(__cplusplus)
```

Comment:
shouldn't it rather be
```suggestion
#elif defined(__GNUC__) && ((__GNUC__ == 4 && __GNUC_MINOR__ >= 6) || __GNUC__ > 4) && !defined(__cplusplus)
```

Reply:
You are right, it seems that something went wrong when I was copying the macro. I already realized it and updated the PR as well. Please have a look again :)